cpp/src/io/orc/stripe_enc.cu

```diff
@@ -1233,7 +1232,7 @@ __global__ void __launch_bounds__(1024)
                        ? statuses[ss.first_block + b].bytes_written
                        : src_len;
       uint32_t blk_size24{};
-      if (dst_len >= src_len) {
+      if (statuses[ss.first_block + b].status == 0) {
```

Comment:
Previous condition was implicit, led to wrong output when the compressed data is larger.